bindings/Fortran/modules/adios2_adios_init_mod_serial.f90

```diff
@@ -2,44 +2,35 @@
 ! Distributed under the OSI-approved Apache License, Version 2.0.  See
 !  accompanying file Copyright.txt for details.
 !
-!  adios2_init_mod.f90 : ADIOS2 Fortran bindings nonMPI init functions
-!
-!   Created on: Mar 13, 2017
-!       Author: William F Godoy godoywf@ornl.gov
+!  adios2_adios_init_mod_serial.f90 : ADIOS2 Fortran bindings for ADIOS
+!                                     class Init functions (serial variants)
 !
 
-module adios2_adios_init_mod
+submodule ( adios2_adios_init_mod ) serial
     use adios2_parameters_mod
     use adios2_functions_mod
     implicit none
 
-    interface adios2_init
-        module procedure adios2_init_nompi
-        module procedure adios2_init_nompi_debug
-        module procedure adios2_init_config
-        module procedure adios2_init_config_debug
-    end interface
-
 contains
 
-    subroutine adios2_init_nompi(adios, adios2_debug_mode, ierr)
+    module subroutine adios2_init_serial(adios, adios2_debug_mode, ierr)
         type(adios2_adios), intent(out) :: adios
         logical, value, intent(in) :: adios2_debug_mode
         integer, intent(out) :: ierr
```

Comment:
The PGI Fortran compiler is unhappy with the use of the `value` attribute on `adios2_debug_mode`.  It complains:

> ```
> PGF90-S-1057-Definition argument name _V_adios2_debug_mode does not match declaration argument name adios2_debug_mode (/.../bindings/Fortran/modules/adios2_adios_init_mod_serial.f90: 16)
>   0 inform,   0 warnings,   1 severes, 0 fatal for adios2_init_serial
> ```

This happens only when it is a `module subroutine`, but making it such is necessary to isolate the implementation in a submodule.

Ideas anyone?

Reply:
I can reproduce this PGI diagnostic in a simple example:

```console
$ cat foo.f90
module foo
  interface
    module subroutine bar(x)
      logical, value, intent(in) :: x
    end subroutine
  end interface
end module

$ cat foo_sub.f90
submodule (foo) sub
contains
  module subroutine bar(x)
    logical, value, intent(in) :: x
  end subroutine
end submodule

$ pgfortran --version
pgfortran 19.10-0 linuxpower target on Linuxpower

$ pgfortran -c foo.f90

$ pgfortran -c foo_sub.f90
PGF90-S-1057-Definition argument name _V_x does not match declaration argument name x (foo_sub.f90: 3)
  0 inform,   0 warnings,   1 severes, 0 fatal for bar
```

Reply:
Can't you work around this by not using value here? And then use an interface block for the actual `bind(C)` function locally, and have `bar()` just be a wrapper that calls the C function?

[I haven't looked at this closely, so I might be off base -- but it seems that the problem is the `value`, and that's only useful in `bind(C)` functions, right?]

Either way, that doesn't help with the lack of support of submodules in older gfortran, though.

Reply:
I've updated the branch to remove the `VALUE` specifier.  I only had it because it was there before, but I don't think we need it.